clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -473,6 +480,17 @@ protected void processConsumerAction(ConsumerAction message, Store store) throws
          */
         endSegment(partition);
         break;
+      case VERSION_ROLE_CHANGE_TO_TRIGGER_RESUBSCRIBE:
+        partitionConsumptionState = partitionConsumptionStateMap.get(partition);
+        if (partitionConsumptionState == null) {
+          LOGGER.info(
+              "Consumer action: {} is skipped for replica: {}, because PCS is null and the partition may have been unsubscribed.",
+              message,
+              Utils.getReplicaId(topicName, partition));
+          return;
+        }
+        resubscribe(partitionConsumptionState, message.getVersionRole());
```

Comment:
This action is quite expensive, and I think we should introduce a feature flag to control this feature, and when the feature flag is off, `SIT` shouldn't even receive such event.
This feature flag will serve two purposes (at least):
1. Gradual ramp.
2. Quick rollback if this feature is problematic.

Reply:
Ideally, this action could only be triggered when version role changed for all store versions. Even it is for all partitions, but it is not very frequent during the store version life cycle.  For safety, we could introduce a host level flag to turn on and off, I will add it.